src/routes/reviewRouter.ts

```diff
@@ -1,7 +1,14 @@
-import { Router } from 'express';
+import { Router } from "express";
+import { validateLoginInput, validateReview } from "../middleware/validateMiddleware";
+import { checkUser, authenticateUser } from "../middleware/authMiddleware";
+import { catchAsync } from "../utils/catchAsync";
+import Review from "../models/Review";
+import { createReview, updateReview, deleteReview } from "../controllers/reviewController";    
 
 const router = Router();
 
-router.get('/')
+router.post('/', validateReview, authenticateUser, catchAsync(createReview));
+router.put('/', validateReview, authenticateUser, catchAsync(updateReview));
```

Comment:
You could add a middleware to check if the current user is the author of the review or not to be able to delete this review. Hint: look at my authenticateUser, I have already pass in a user id through the req object.